apps/aehyperchains/src/aehc_connector.erl

```diff
@@ -0,0 +1,98 @@
+-module(aehc_connector).
+
+-export([send_tx/1, get_block/1]).
+
+-export([tx/3, header/2, block/2]).
+
+-export([publish/2, subscribe/1]).
+
+-type connector() :: atom().
+
+-callback send_tx(Tx::aetx:tx()) -> binary().
+
+-callback get_block(Num::integer()) -> block().
+
+-export_type([tx/0, header/0, block/0]).
+
+-spec connector() -> connector().
+connector() ->
+    ok.
+
+%%%===================================================================
+%%%  Parent chain simplified proto
+%%%===================================================================
+
+-record(tx, {
+             sender :: binary(),
+             recipient :: binary(),
+             payload :: binary()
+            }).
+
+-type tx() :: #tx{}.
+
+-record(header, {
+                 hash :: binary(),
+                 number = 0 :: integer()
+                }).
+
+-type header() :: #header{}.
+
+-record(block, {
+                header :: header(),
+                txs :: [tx()]
+               }).
+
+-type block() :: #block{}.
+
+-spec tx(Sender::binary(), Recipient::binary(), Payload::binary()) -> tx().
+tx(Sender, Recipient, Payload) when 
+      is_binary(Sender), is_binary(Recipient), is_binary(Payload) ->
+    #tx{ sender=Sender, recipient=Recipient, payload=Payload }.
+
+-spec header(Hash::binary(), Num::integer()) -> header().
+header(Hash, Num) when 
+      is_binary(Hash), is_integer(Num) ->
+    #header{ hash=Hash, number=Num }.
+
+-spec block(Header::header(), Txs::[tx()]) -> block().
+block(Header, Txs) when
+      is_record(Header, header), is_list(Txs) ->
+    #block{ header=Header, txs=Txs }.
+
+%%%===================================================================
+%%%  Parent chain interface
+%%%===================================================================
+
+-spec send_tx(Tx::aetx:tx()) ->
+                    {ok, TxHash::binary()} | {error, {term(), term()}}.
+send_tx(Tx) ->
```

Comment:
Why so low level? Imo we should provide interface just for commitments, we are not interested in arbitrary transactions

Reply:
Yes, that's correct (also got short discussion with @gorbak25 about, there is also could be contract call execution result)